v3/src/data-interactive/handlers/attribute-handler.ts

```diff
@@ -46,10 +79,49 @@ export const diAttributeHandler: DIHandler = {
     }
     return attributeNotFoundResult
   },
-  create: diNotImplementedYet,
+  create(resources: DIResources, values?: DIValues) {
+    const { dataContext } = resources
+    if (!dataContext) return dataContextNotFoundResult
+    const metadata = getSharedCaseMetadataFromDataset(dataContext)
+
+    // Wrap single attribute in array and bail if any new attributes are missing names
+    const attributeValues = Array.isArray(values) ? values : [values]
+    const attributeErrors = attributeValues.map(singleValue => {
+      if (!singleValue?.name) {
+        return {
+          success: false,
+          values: { error: t("V3.DI.Error.fieldRequired", { vars: ["Create", "attribute", "name"] }) }
+        } as const
+      }
+      return { success: true }
+    }).filter(error => !error.success)
+    if (attributeErrors.length > 0) return attributeErrors[0]
+
+    // Create the attributes
+    const attributes: IAttribute[] = []
+    const createAttribute = (value: DISingleValues) => {
+      dataContext.applyUndoableAction(() => {
+        withoutUndo()
```

Comment:
As mentioned elsewhere, we need to come up with a consistent strategy for dealing with undo/redo of plugin-driven actions. Since first reviewing this PR I had a meeting with @eireland at which we determined that in v2 some plugin-driven actions are undoable and some are not, depending entirely on the vagaries of the implementation. So there isn't a consistent v2 strategy to follow. Need not hold up this PR, however.